state/watcher.go

```diff
@@ -2168,26 +2178,28 @@ func (w *documentFieldWatcher) initial() error {
 
 	field := w.initialKnown
 
-	var data map[string]interface{}
+	data := w.document()
 	if err := col.Find(w.members).One(&data); err == nil {
 		if newField, err := w.extract(data); err == nil {
 			field = newField
 		}
 	}
-	w.known = field
+	w.known = &field
 
 	logger.Tracef("Started watching %s for %v: %q", w.collection, w.members, field)
 	return nil
 }
 
 func (w *documentFieldWatcher) merge(change watcher.Change) (bool, error) {
-	if change.Revno == -1 {
+	// we care about change.Revno equalling -1 as we want to know about
+	// documents being deleted.
+	if w.known == nil {
```

Comment:
I think this fix is the wrong way around.
'w.known == nil' means that we think the document is deleted.
I think you want something more like:
```
if change.Revno == -1 {
 // treat this as the document being deleted
 if w.known != nil {
   w.known = nil
   return true, nil
 }
 return false, nil
}
```

Reply:
ah right, nps.